Brand new arays doc. A small amount of work remains to be done. (#4307)
yugabyte/yugabyte-db: docs/content/latest/api/ysql/datatypes/type_array/array-constructor.md

- Array-like functionality is essential in any programming language.
- The `array[]` constructor is most valuable when the expressions that it uses are composed using declared variables, and especially formal parameters, that are used to build whatever values are intended. In this example, the values have the user-defined data type `rt`. In other words, the `array[]` constructor is particularly valuable when you build an array programmatically from scalar values that you know first at run time.
- It vividly demonstrates the semantic effect of the `array[]` constructor like this:
```
declare
  r     rt[];
  two_d rt[];
begin
  ...
  assert (array_dims(r) = '[1:3]'), 'assert failed';
  one_d_1 := array[r[1], r[2], r[3]];
  assert (one_d_1 = r), 'assert failed';
```
The `array_dims()` function is documented [here](../functions-operators/properties/#array-dims).

Run this to create the required user-defined _"row"_ type and the table function and then to invoke it.

```postgresql

> -- Don't create "type rt" if it's still there followng the previous example.

Author: Damn. I just checked this carefully. It seems that Typora.app doesn't spell check within triple-tick blocks. Must be an el-cheapo compromise 'cos so many of the words that you find in code blocks aren’t in the English dictionary. My favorite Pycharm knows about Python syntax, of course, and so it spell-checks within comments, no matter what marking among several you use to mark distinguish them as such. Did you find this with a spell checker or by eye?

---
title: The array[] value constructor
linkTitle: array[] constructor
headerTitle: The array[] value constructor
description: The array[] value constructor
menu:
  latest:
    identifier: array-constructor
    parent: api-ysql-datatypes-array
    weight: 10

> isTocNested: false

Reviewer: ```suggestion
isTocNested: true
```

> isTocNested: false
> showAsideToc: false

Reviewer: ```suggestion
isTocNested: true
showAsideToc: true
```

---

> The `array[]` value constructor is a special variadic function. Uniquely among all the functions described in the present major _"[Array data types and functionality](../../type_array/)"_ section, it uses square brackets to surround its list of actual arguments.

Reviewer: ```suggestion
The `array[]` value constructor is a special variadic function. Uniquely among all the functions described in this [Array data types and functionality](../../type_array/) section, it uses square brackets (`[]`) to surround its list of actual arguments.
```

## Purpose and signature

> **Purpose:** create an array value from scratch using an expression for each of the array's values. Such an expression can itself use the `array[]` constructor or an _[array literal](../literals/)_. 

Reviewer: ```suggestion
**Purpose:** Create an array value from scratch using an expression for each of the array's values. Such an expression can itself use the `array[]` constructor or an [array literal](../literals/). 
```

**Signature** 
```
input value:       [anyarray | [ anyelement, [anyelement]* ]
return value:      anyarray
```
**Notes:** You can meet the goal of creating an array from directly specified values, instead, by using an _[array literal](../literals/)_.

These thee ordinary functions also create an array value from scratch:

> - The _[array_fill()](../functions-operators/array-fill/)_ builtin SQL function creates a "blank canvas" array of the specified shape with all values set the same to what you want.

Author: Same here too. Clicking "Resolve converstation".

> - The _[array_agg()](../functions-operators/array-agg-unnest/#array-agg)_ builtin SQL function creates an array (of, in general, an implied _"row"_ type) from a SQL subquery.

Reviewer: ```suggestion
- The _[array_agg()](../functions-operators/array-agg-unnest/#array-agg)_ built-in SQL function creates an array (of, in general, an implied _"row"_ type) from a SQL subquery.
```

Reviewer: I'm leaning towards using "predefined" instead of "built-in," but need to think a bit more about it.

Author: This one (change "builtin" to built-in" which I did globally before the push that you watched me do) has brought the "outdated" status and the "commit" button is disabled. My only option seems to be to ignore it. There's no way for me to say that it's already fixed.

Author: So I'm clicking "Resolve conversation".
